AnkiDroid/src/main/java/com/ichi2/anki/MediaRegistration.java

```diff
@@ -76,10 +79,28 @@ public String loadImageIntoCollection(Uri uri) throws IOException {
             fileName = fileNameAndExtension.getKey();
         }
 
-        File clipCopy = File.createTempFile(fileName, fileNameAndExtension.getValue());
-        String tempFilePath = clipCopy.getAbsolutePath();
-        long bytesWritten = CompatHelper.getCompat().copyFile(fd, tempFilePath);
+        File clipCopy;
+        long bytesWritten;
+
+        if (!fileNameAndExtension.getValue().equals(".png") && CollectionHelper.getInstance().getCol(mContext).getConf().optBoolean("pastePNG")) {
```

Comment:
`.png` is guaranteed to be non-null

```suggestion
        if (!".png".equals(fileNameAndExtension.getValue()) && CollectionHelper.getInstance().getCol(mContext).getConf().optBoolean("pastePNG")) {
```